akka-stream-tests/src/test/scala/akka/stream/scaladsl/One2OneBidiFlowSpec.scala

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import org.scalactic.ConversionCheckedTripleEquals
+import akka.stream.ActorMaterializer
+import akka.stream.testkit.AkkaSpec
+import akka.stream.testkit._
+
+class One2OneBidiFlowSpec extends AkkaSpec with ConversionCheckedTripleEquals {
+  import FlowGraph.Implicits._
+
+  implicit val mat = ActorMaterializer()
+
+  /**
+   * Wraps a flow with one-to-one guarding, which
+   *
+   * 1. triggers an `OutputTruncationException` if the wrapped stream terminates
+   *    before exactly one output has been produced for each input element
+   * 2. Backpressures surplus output elements (i.e. ones that are being produced without
+   *    having a corresponding input element)
+   * 3. makes sure that output elements are being produced in lockstep with input elements,
+   *    i.e. no output element is ever being produced before the corresponding input element hasn't been received
+   *
+   * // TODO: make this available on Flows; Where do we place it best?
```

Comment:
Hm, that's a weird one... We don't have similar things to it yet, maybe some new place hm.

// cc @drewhk @rkuhn